docs/Running-Spark-on-Tachyon.md

```diff
@@ -35,97 +39,128 @@ recompile Spark with the right version of tachyon-client by changing the version
-later and has set up Tachyon and Hadoop in accordance to these guides
+If the version of Spark is not supported by your Tachyon installation by default (e.g., you are
+trying out the latest Tachyon release with some older Spark installation), one can recompile Spark
+by updating the correct version of tachyon-client in Spark dependency. To do that, edit
+`spark/core/pom.xml` and change the dependency version of `tachyon-client` to 
+`target_tachyon_version`:
+
+```xml
+<dependency>
+  <groupId>org.tachyonproject</groupId>
+  <artifactId>tachyon-client</artifactId>
+  <version>target_tachyon_version</version>
+  ...
+</dependency>
+```
+
+## Prerequisites
+
+* Make sure your Spark installation is compatible with Tachyon installation. Check
+[Compatibility](#compatibility) session for more instructions.
+* Tachyon cluster has been set up in accordance to these guides for either
 [Local Mode](Running-Tachyon-Locally.html) or [Cluster Mode](Running-Tachyon-on-a-Cluster.html).
-
-If you are running a Spark version less than 1.0.0, please add the following line to
+* If Spark version is earlier than `1.0.0`, please add the following line to
 `spark/conf/spark-env.sh`.
 
-    export SPARK_CLASSPATH=/pathToTachyon/client/target/tachyon-client-{{site.TACHYON_RELEASED_VERSION}}-jar-with-dependencies.jar:$SPARK_CLASSPATH
+```bash
+export SPARK_CLASSPATH=/pathToTachyon/client/target/tachyon-client-{{site.TACHYON_RELEASED_VERSION}}-jar-with-dependencies.jar:$SPARK_CLASSPATH
+```
+
+* If Tachyon is run on top of a Hadoop 1.x cluster, create a new file `spark/conf/core-site.xml`
+with the following content:
+
+```xml
+<configuration>
+  <property>
+    <name>fs.tachyon.impl</name>
+    <value>tachyon.hadoop.TFS</value>
+  </property>
+</configuration>
+```
 
-If running a hadoop 1.x cluster, create a new file `spark/conf/core-site.xml` Add the following to it:
 
-    <configuration>
-      <property>
-        <name>fs.tachyon.impl</name>
-        <value>tachyon.hadoop.TFS</value>
-      </property>
-    </configuration>
+* If you are running tachyon in fault tolerant mode with zookeeper and the Hadoop cluster is a 1.x, 
+add the following additionally entry to the previously created `spark/conf/core-site.xml`:
 
-Put a file X into HDFS and run the Spark shell:
+```xml
+<property>
+  <name>fs.tachyon-ft.impl</name>
+  <value>tachyon.hadoop.TFSFT</value>
+</property>
+```
 
-    $ ./spark-shell
-    $ val s = sc.textFile("tachyon://localhost:19998/X")
-    $ s.count()
-    $ s.saveAsTextFile("tachyon://localhost:19998/Y")
+and the following line to `spark/conf/spark-env.sh`:
 
-Take a look at [http://localhost:19999](http://localhost:19999). There should be an output file
-`Y` which contains the number of words in the file `X`.
-Put a file X into HDFS and run the Spark shell:
+```bash
+export SPARK_JAVA_OPTS="
+  -Dtachyon.zookeeper.address=zookeeperHost1:2181,zookeeperHost2:2181 \
+  -Dtachyon.usezookeeper=true \
```

Comment:
This property is changed into `tachyon.zookeeper.enabled` in current master branch.